src/model.js

```diff
@@ -3844,9 +3287,11 @@ Instead of specifying a Model, either:
       originalValue = this.dataValues[key];
     }
 
+    const attributeDefinition = modelDefinition.attributes.get(key);
+
     // If not raw, and there's a custom setter
-    if (!options.raw && this._customSetters[key]) {
-      this._customSetters[key].call(this, value, key);
+    if (!options.raw && attributeDefinition?.set) {
+      attributeDefinition.set.call(this, value, key);
       // custom setter should have changed value, get that changed value
       // TODO: v5 make setters return new value instead of changing internal store
```

Comment:
What a nice TODO for v5 :)

Reply:
I also don't really understand what it means 😅